nncf/tensorflow/graph/converter.py

```diff
@@ -87,24 +59,132 @@ def convert_keras_model_to_nncf_graph(model: tf.keras.Model) -> NNCFGraph:
     def __init__(self, model: tf.keras.Model):
         self._model = model
+        self._node_info = {}  # type: Dict[str, Dict]
+        self._custom_layer_infos = self._collect_custom_layer_infos(self._model)
+        self._nncf_node_names_vs_custom_layer_name = {}  # type: Dict[NNCFNodeName, str]
+
+    def _collect_custom_layer_infos(self, model: tf.keras.Model,
+                                    use_graph_var_names: bool = False) -> Dict[str, CustomLayerInfo]:
+        custom_layers = TFModelConverter.get_custom_layers(model)
+        retval = {}
+        for layer_name, layer in custom_layers.items():
+            layer_input_spec = [tf.TensorSpec.from_tensor(tensor)
+                                for tensor in layer.input] if isinstance(layer.input, list) \
+                else tf.TensorSpec.from_tensor(layer.input)
+
+            # TODO: Use the custom layer's inbound_nodes/outbound_nodes to determine what edges
+            #       should connect it to the rest of the graph. Currently the custom layer
+            #       subgraph will be present in the NNCFGraph after conversion, which is useful
+            #       for purposes of weight modification target point creation and usage,
+            #       but the subgraph won't be connected to the rest of the graph; in the main graph
+            #       component, the custom layer will still be represented by a single node
+            concr_fn = tf.function(layer).get_concrete_function(layer_input_spec, training=False)
+            wrapped_function = convert_variables_to_constants_v2(concr_fn, lower_control_flow=False)
+
+            graphdef_nodes = wrapped_function.graph.as_graph_def().node
+            graphdef_name_to_layer_var_map = {} if use_graph_var_names else \
+                TFModelConverter._get_graphdef_name_to_layer_var_map(concr_fn)
+            nodes = {graphdef_name_to_layer_var_map.get(node.name, node.name): node for node in graphdef_nodes}
+            graphdef_node_name_vs_node = {node.name: node for node in graphdef_nodes}
+
+            custom_layer_info = CustomLayerInfo()
+            for pretty_node_name, node in nodes.items():
+                custom_layer_info.graphdef_node_name_to_pretty_node_name[node.name] = pretty_node_name
+
+            for pretty_node_name, node in nodes.items():
+                weight_node_name = None
+                metatype = get_op_metatype(node.op)
+                if metatype in WEIGHTABLE_TF_OP_METATYPES:
+                    graphdef_weight_node_name = self._get_graphdef_node_name_for_custom_layer_node_weight(
+                        node,
+                        graphdef_node_name_vs_node)
+                    if graphdef_weight_node_name in graphdef_name_to_layer_var_map:
+                        weight_node_name = graphdef_name_to_layer_var_map[graphdef_weight_node_name]
+                    else:
+                        nncf_logger.warning("Could not associate a weighted custom layer node {} "
```

Comment:
Could you please use `'` instead of `"` here and below?

Reply:
Done